src/Microsoft.SqlTools.ServiceLayer/ShowPlan/ShowPlanGraph/XmlPlanNodeBuilder.cs

```diff
@@ -371,6 +374,100 @@ private IEnumerable<BaseStmtInfoType> EnumStatements(StmtBlockType statementBloc
+            var nsmgr = new XmlNamespaceManager(stmtXmlDocument.NameTable);
+            nsmgr.AddNamespace("shp", "http://schemas.microsoft.com/sqlserver/2004/07/showplan");
+            XmlNode rootNode = stmtXmlDocument.DocumentElement;
+
+            XmlNode missingIndexes = rootNode.SelectSingleNode("descendant::shp:MissingIndexes", nsmgr);
+            //Returning null if no missing indexes are found in the xml file
+            if (missingIndexes == null)
+            {
+                return null;
+            }
+            XmlNodeList indexGroups = missingIndexes.SelectNodes("descendant::shp:MissingIndexGroup", nsmgr);
+
+            bool memoryOptimzed = false;
+            XmlNode scan = rootNode.SelectSingleNode("descendant::shp:IndexScan", nsmgr);
+            if (scan == null)
+            {
+                scan = rootNode.SelectSingleNode("descendant::shp:TableScan", nsmgr);
+            }
+            if (scan != null && scan.Attributes["Storage"] != null)
+            {
+                if (0 == string.Compare(scan.Attributes["Storage"].Value, "MemoryOptimized", StringComparison.Ordinal))
+                {
+                    memoryOptimzed = true;
+                }
+            }
+
+            const string createIndexTemplate = "CREATE NONCLUSTERED INDEX [<Name of Missing Index, sysname,>]\r\nON {0}.{1} ({2})\r\n";
+            const string addIndexTemplate = "ALTER TABLE {0}.{1}\r\nADD INDEX [<Name of Missing Index, sysname,>]\r\nNONCLUSTERED ({2})\r\n";
+            const string includeTemplate = "INCLUDE ({0})";
+
+            List<MissingIndex> parsedIndexes = new List<MissingIndex>();
+            foreach (XmlNode indexGroup in indexGroups)
+            {
+                XmlNode missingIndex = indexGroup.SelectSingleNode("descendant::shp:MissingIndex", nsmgr);
+                string database = missingIndex.Attributes["Database"].Value;
+                string schemaName = missingIndex.Attributes["Schema"].Value;
+                string tableName = missingIndex.Attributes["Table"].Value;
+                string indexColumns = string.Empty;
+                string includeColumns = string.Empty;
+
+                // populate index columns and include columns
+                XmlNodeList columnGroups = missingIndex.SelectNodes("shp:ColumnGroup", nsmgr);
+                foreach (XmlNode columnGroup in columnGroups)
+                {
+                    foreach (XmlNode column in columnGroup.ChildNodes)
+                    {
+                        string columnName = column.Attributes["Name"].Value;
+                        if (0 != string.Compare(columnGroup.Attributes["Usage"].Value, "INCLUDE", StringComparison.Ordinal))
+                        {
+                            if (indexColumns == string.Empty)
+                                indexColumns = columnName;
+                            else
+                                indexColumns = string.Format("{0},{1}", indexColumns, columnName);
```

Comment:
Can also use string interpolation: `indexColumns = $"{indexColumns},{columnName}"`, slightly more compact way of doing the same thing.